### PR TITLE
feature: ui

### PR DESCRIPTION
**Pull request by bokuhe**
UI
- setting screen and merge conflict
- add popup modal (pin reset, signout)
- pin retry count & lock

### CODE REVIEW

File: src/hooks/auth/useAuth.ts [hunk 1]
```diff
@@ -45,9 +45,11 @@ export type UseAuthReturn = {
 const useAuth = (): UseAuthReturn => {
   const [user, setUser] = useRecoilState(appStore.user)
   const { connect, disconnect } = useConnection()
-  const { setCurrentUser } = useSendbirdChat()
+  const { setCurrentUser, features } = useSendbirdChat()
```
Comment (denniswon) [inline src/hooks/auth/useAuth.ts:48]: this should be done in useNotificationRegister

Comment (denniswon) [reply src/hooks/auth/useAuth.ts:48]: And we should default to on once the user grants permission. And if the user turns off, degister from sendbird, remove from firestore, remove async storage, and also remove permission

**Review by denniswon — CHANGES_REQUESTED** [review]
And we should default to on once the user grants permission. And if the user turns off, degister from sendbird, remove from firestore, remove async storage, and also remove permission

File: src/hooks/auth/useAuth.ts (hunk 1 above)
Comment (bokuhe) [reply src/hooks/auth/useAuth.ts:48]: Now the token register is running in two functions.
useNotificationRegister hook and connect function of useConnection hook.

useConnection is a library of sendbird, when connect call, token is automatically registered.
so, autoRegistration was changed to false where connect hook is used.

and it is already trying to register in useNotificationRegister when the app is running. useNotificationRegister is used even when the value changes in Setting.

File: src/components/atoms/FormModal.tsx [hunk 2]
```diff
@@ -1,16 +1,115 @@
...
-const FormModal = (props: ModalProps): ReactElement => {
-  const { children, ...rest } = props
+const FormModal = (
+  props: ModalProps & {
+    title: string
+    message: string
+    positive?: string
+    positiveCallback?: () => void
+    negative?: string
+    negativeCallback?: () => void
```
Comment (denniswon) [inline src/components/atoms/FormModal.tsx:19]: just add this to ModalProps. Seems generic. Also, would be better to make a type for modal button type `(text: string, callback: () => void)`, and make `positive?: ModalButton, negative?: ModalButton`

File: src/components/sendbird/ChannelMembersPreview.tsx [hunk 3]
```diff
@@ -24,39 +24,42 @@ const ChannelMembersPreview = React.memo(
...
 
     return (
-      <Row
-        style={[
-          styles.container,
-          containerStyle,
-          { paddingStart: (-1 * size) / 2 },
-        ]}
-      >
+      <Row style={[styles.container, containerStyle]}>
```
Comment (denniswon) [inline src/components/sendbird/ChannelMembersPreview.tsx:27]: can you post the screenshot of how this looks? Please verify that this Component looks good for "BOTH" chatchard and channel settings screen

File: src/hooks/independent/useNotificationConf.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import { AuthorizationStatus } from '@notifee/react-native'
+import AsyncStorage from '@react-native-async-storage/async-storage'
+import messaging from '@react-native-firebase/messaging'
+import { UTIL } from 'consts'
+import { LocalStorageKey } from 'types'
+
+const useNotificationConf = (): {
+  isEnableNotification: () => Promise<boolean>
```
Comment (denniswon) [inline src/hooks/independent/useNotificationConf.ts:8]: nit: grammar `enableNotification` or `notificatinEnabled` makes more sense

File: src/hooks/independent/useNotificationRegister.ts [hunk 5]
```diff
@@ -0,0 +1,129 @@
+import useAuth from 'hooks/auth/useAuth'
```
Comment (denniswon) [inline src/hooks/independent/useNotificationRegister.ts:1]: We should create a folder `src/hooks/notification` and put all the hooks there

File: src/screens/PinScreen.tsx [hunk 6]
```diff
@@ -186,11 +205,42 @@ const PinScreen = (): ReactElement => {
...
+
+    return () => {
+      setVisibleResetModal(false)
+    }
+  })
+
   useEffect(() => {
     clearInputPin()
     resetNewPin()
+
```
Comment (denniswon) [inline src/screens/PinScreen.tsx:236]: remove blank line

File: src/screens/channels/ChannelListingsScreen/index.tsx.tsx [hunk 7]
```diff
@@ -46,11 +56,33 @@ const Contents = ({ channelUrl }: { channelUrl: string }): ReactElement => {
...
+          {t('Nft.ChannelListingNoListedNft')}
         </FormText>
-      </View>
+        <FormButton
+          figure="outline"
+          containerStyle={styles.buttonListNft}
+          textStyle={{ color: COLOR.black._300 }}
+          rightIcon="ios-chevron-forward"
+          onPress={(): void => {
+            // navigation.navigate(Routes.ListNft, { channelUrl })
```
Comment (denniswon) [inline src/screens/channels/ChannelListingsScreen/index.tsx.tsx:80]: what is this?

**Review by denniswon — CHANGES_REQUESTED** [review]
Overall LGTM. Requested changes per comment.

Also, from next time, can you create separate PRs for different work/features? Hard to review

File: src/screens/channels/ChannelListingsScreen/index.tsx.tsx (hunk 7 above)
Comment (bokuhe) [reply src/screens/channels/ChannelListingsScreen/index.tsx.tsx:80]: 스크린샷 2023-06-28 오후 2 45 59

This button lists the nft when there is no nft listed.
Implementation of list nft view was not completed.
 I'll remove the button.

File: src/components/sendbird/ChannelMembersPreview.tsx (hunk 3 above)
Comment (bokuhe) [reply src/components/sendbird/ChannelMembersPreview.tsx:27]: Design
스크린샷 2023-06-28 오후 2 54 59

App
스크린샷 2023-06-28 오후 2 57 44

**Review by denniswon — APPROVED** [review]

**Pull request merged**